src/rest/DataProduceEndpoints.js

```diff
@@ -60,13 +60,16 @@ module.exports = (streamFetcher, publisher, volumeLogger = new VolumeLogger(0))
             // req.stream is written by authentication middleware
             publisher.publish(
                 req.stream,
+                publisher.getStreamPartition(req.stream, req.query.pkey),
                 timestamp,
+                0, // sequenceNumber
```

Comment:
Can we add read the missing query params? (and if undefined, pass default empty value)

- `req.query.seq`
- `req.query.prev_ts`
- `req.query.prev_seq`

Not necessarily pretty but feature-complete.